src/aind_hcr_data_transformation/compress/czi_to_zarr_s3.py

```diff
@@ -222,7 +223,8 @@ def create_downsample_dataset(
 
     down_dataset = ts.open(down_spec).result()
     downsampled_data = downsampled_dataset.read().result()
-    down_dataset.write(downsampled_data).result()
+    with ts.Transaction() as transaction:
```

Comment:
Maybe we should wrap the read above in a transaction? Alternatively, it might be helpful to at least log a message to check that the read isn't getting stuck.

Reply:
Ah yeah thats a good idea!